src/test/java/com/github/commonsrdf/api/AbstractRDFTermFactoryTest.java

```diff
@@ -0,0 +1,349 @@
+	}
+
+	public abstract RDFTermFactory createFactory();
+
+	@Test
+	public void createGraph() {
+		Graph graph;
+		try {
+			graph = factory.createGraph();
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException(ex);
+			return;
+		}
+
+		assertEquals("Graph was not empty", 0, graph.size());
+		graph.add(factory.createBlankNode(),
+				factory.createIRI("http://example.com/"),
+				factory.createBlankNode());
+
+		Graph graph2 = factory.createGraph();
+		assertNotSame(graph, graph2);
+		assertEquals("Graph was empty after adding", 1, graph.size());
+		assertEquals("New graph was not empty", 0, graph2.size());
+	}
+
+	@Test
+	public void createIRI() throws Exception {
+		IRI example;
+		try {
+			example = factory.createIRI("http://example.com/");
+		} catch (UnsupportedOperationException ex) {
+			Assume.assumeNoException("createIRI not supported", ex);
+			return;
+		}
+
+		assertEquals("http://example.com/", example.getIRIString());
+		assertEquals("<http://example.com/>", example.ntriplesString());
+
+		IRI term = factory.createIRI("http://example.com/vocab#term");
+		assertEquals("http://example.com/vocab#term", term.getIRIString());
+		assertEquals("<http://example.com/vocab#term>", term.ntriplesString());
+
+		// and now for the international fun!
+
+		IRI latin1 = factory.createIRI("http://accént.example.com/première");
+		assertEquals("http://accént.example.com/première",
+				latin1.getIRIString());
+		assertEquals("<http://accént.example.com/première>",
+				latin1.ntriplesString());
+
+		IRI cyrillic = factory.createIRI("http://example.испытание/Кириллица");
+		assertEquals("http://example.испытание/Кириллица",
+				cyrillic.getIRIString());
+		assertEquals("<http://example.испытание/Кириллица>",
+				cyrillic.ntriplesString());
+
+		IRI deseret = factory.createIRI("http://𐐀.example.com/𐐀");
+		assertEquals("http://𐐀.example.com/𐐀", deseret.getIRIString());
+		assertEquals("<http://𐐀.example.com/𐐀>", deseret.ntriplesString());
+	}
+
+	@Test
+	public void createIRIRelative() throws Exception {
+		// Although relative IRIs are defined in
+		// http://www.w3.org/TR/rdf11-concepts/#section-IRIs
+		// it is not a requirement for an implementation to support
+		// it (all instances of an relative IRI should eventually
+		// be possible to resolve to an absolute IRI)
+		try {
+			factory.createIRI("../relative");
+		} catch (UnsupportedOperationException | IllegalArgumentException ex) {
+			Assume.assumeNoException(ex);
+			return;
+		}
+		IRI relative = factory.createIRI("../relative");
+		assertEquals("../relative", relative.getIRIString());
+		assertEquals("<../relative>", relative.ntriplesString());
+
+		IRI relativeTerm = factory.createIRI("../relative#term");
+		assertEquals("../relative#term", relativeTerm.getIRIString());
+		assertEquals("<../relative#term>", relativeTerm.ntriplesString());
+
+		IRI emptyRelative = factory.createIRI(""); // <> equals the base URI
```

Comment:
There *could* be a base URI depending on how the factory was created - which is out of scope for us. But this test bails out early if it is not supported, without raising it as an error. If it is supported, though, then the rest should work.

Jena's `IRIFactory` for instance can be created with a base URI (and in fact, there even an IRIImpl is a IRIFactory) :)